File: frontend/src/api/erpConfig.js
import { api } from './client';

export const erpConfigApi = {
  listar: () => api.get('/api/erp-config'),
  actualizar: (id, data) => api.put(`/api/erp-config/${id}`, data),

  // Subartículos que requieren muestreo (consultados por el agente de
  // detección automática de IR -- ver app/services/agente_muestreo.py).
  // El listado es el catálogo GIT59SAR del ERP combinado con el estado de
  // configuración en LIMSS; el guardado es un upsert por CODSAR (no por id).
  listarSubarticulos: () => api.get('/api/erp-config/subarticulos'),
  guardarSubarticulo: (erpCodsar, data) =>
    api.put(`/api/erp-config/subarticulos/${encodeURIComponent(erpCodsar)}`, data),
};

File: backend/app/schemas/erp_config.py
from pydantic import BaseModel, Field
from typing import Optional
from datetime import datetime


class ErpConfig(BaseModel):
    id: int
    clave: str
    valor: str
    descripcion: Optional[str] = None
    editable: bool
    fecha_modificacion: Optional[datetime] = None
    id_usuario_modificacion: Optional[int] = None


class ErpConfigUpdate(BaseModel):
    valor: str = Field(..., min_length=1, max_length=100)
    descripcion: Optional[str] = Field(None, max_length=200)


class SubarticuloConfig(BaseModel):
    """Una fila por CODSAR del catálogo GIT59SAR del ERP (universo chico de
    subartículos, no confundir con GIM21ART -- los artículos individuales,
    miles de filas), combinada con su estado de configuración en LIMSS si
    ya existe (ver GET /api/erp-config/subarticulos)."""
    erp_codsar: str
    erp_dessar: Optional[str] = None
    configurado: bool
    requiere_muestreo: bool
    id: Optional[int] = None
    fecha_carga: Optional[datetime] = None
    id_usuario_carga: Optional[int] = None


class SubarticuloUpsert(BaseModel):
    requiere_muestreo: bool
